tools/cli/adminCommands.go

```diff
@@ -209,6 +213,141 @@ func AdminListNamespaces(c *cli.Context) {
 	paginate(c, paginationFunc)
 }
 
+// UpdateNamespace updates a namespace
+func UpdateNamespace(c *cli.Context) error {
+	namespace := getRequiredGlobalOption(c, FlagNamespace)
+
+	client := cFactory.FrontendClient(c)
+
+	var req *adminservice.UpdateNamespaceRequest
+	ctx, cancel := newContext(c)
+	defer cancel()
+
+	if c.IsSet(FlagActiveClusterName) {
+		activeCluster := c.String(FlagActiveClusterName)
+		fmt.Printf("Will set active cluster name to: %s, other flag will be omitted.\n", activeCluster)
+		replicationConfig := &replicationpb.NamespaceReplicationConfig{
+			ActiveClusterName: activeCluster,
+		}
+		req = &adminservice.UpdateNamespaceRequest{
+			Namespace:         namespace,
+			ReplicationConfig: replicationConfig,
+		}
+	} else {
+		resp, err := client.DescribeNamespace(ctx, &workflowservice.DescribeNamespaceRequest{
+			Namespace: namespace,
+		})
+		if err != nil {
+			if _, ok := err.(*serviceerror.NotFound); !ok {
+				return fmt.Errorf("namespace update failed: %s", err)
+			} else {
+				return fmt.Errorf("namespace %s does not exist: %s", namespace, err)
+			}
+		}
```

Comment:
1. this logic should be handled by server?

the API is called UpdateNamespace, meaning caller should only send out variables to be updated. (send out diff)
if the API is called SetNamespace/PutNamespace, then caller is responsible for telling the sever exactly how a namespace should looks like. (send out entire namespace)

there are some limitation of the update API, e.g. hard to specify `delete one entry from a slice map`

so what is the intention? update or set / put?